pallets/pallet-asset-switch/src/tests/switch.rs

```diff
@@ -320,12 +327,37 @@ fn fails_on_not_enough_user_local_balance() {
 		.with_balances(vec![(user.clone(), 100_000, 0, 1)])
 		.with_switch_pair_info(NewSwitchPairInfo {
 			circulating_supply: 0,
-			pool_account,
+			pool_account: pool_account.clone(),
 			remote_asset_id: REMOTE_ERC20_ASSET_ID.into(),
 			remote_fee: XCM_ASSET_FEE.into(),
 			remote_reserve_location: ASSET_HUB_LOCATION.into(),
 			status: SwitchPairStatus::Running,
 			total_issuance: 100_000,
+			min_remote_balance: 0,
+		})
+		.build()
+		.execute_with(|| {
+			assert_noop!(
+				Pallet::<MockRuntime>::switch(
+					RawOrigin::Signed(user.clone()).into(),
+					100_000,
+					Box::new(ASSET_HUB_LOCATION.into())
+				),
+				Error::<MockRuntime>::UserSwitchBalance
+			);
+		});
```

Comment:
Why is this test case failing?

Reply:
Because the user has 1 token that is in a hold state, hence they cannot perform the switch. Same for the case below it, which uses reserves.